Matrix.py

```diff
@@ -0,0 +1,175 @@
+ mul, scalar_mul and division by scalar methods and non- mathemathic methods
+ as repr, str, hash and eq."""
+
+
+class Matrix:
+    """A matrices calculator for basic math actions
+        Attributes:
+            data(tuple of tuples): a Matrix- each line is a tuple in the major tuple """
+
+    def __init__(self, data):
+        #  inits a Matrix from received data (type= tuple), throws error for non- tuple inputs.
+        check_value = True
+        if type(data) is tuple:
+            for tup in data:
+                if len(tup) != len(data) or type(tup) is not tuple:
+                    check_value *= 0
+        else:
+            check_value *= 0
+        if check_value:
+            self.data = data
+        else:
+            raise ValueError("Invalid matrix")
+
+    def __iter__(self):
+        return iter(self.data)
+
+    def __getitem__(self, item):
+        return self.data[item]
+
+    def __repr__(self):
+        """prints the matrix in a format that can be used for build another Matrix"""
+        return f"Matrix({self.data})"
+
+    # str returns the Matrix in the mathemathic format
+    def __str__(self):
+        matrix_in_lines = ""
+        for i in range(len(self.data)):
+            matrix_in_lines += (str(self.data[i]) + "\n")
+        return matrix_in_lines
+
+    # unity method creates a matrix of zeros with a diagonal of ones (unity is the Matrix's size)
+    @staticmethod
+    def unity(unit):
+        major_list = [[1 if i == index else 0 for i in range(unit)] for index in range(unit)]
+        sub_list = [tuple(lt) for lt in major_list]
+        # converts list of lists to type Matrix. lt are the inside lists
+        matrix_to_return = Matrix((tuple(sub_list)))
+        return matrix_to_return
+
+    # ones method returns a Matrix full of ones which sizes determined by "unit"
+    @staticmethod
+    def ones(unit):
+        sub_list = [tuple(lt) for lt in [([1] * unit) for i in range(unit)]]
+        # convert list of lists to type Matrix. lt are the inside lists
+        matrix_to_return = Matrix((tuple(sub_list)))
+        return matrix_to_return
+
+    def __add__(self, other):
+        """add a matrix to another one
+            args: self.data (tuple)- A matrix
+                other(tuple) -A matrix
+
+            returns: return the new matrix (tuple)
+
+             raises: syntaxerror if "other" isn't a matrix"""
+        if type(other) is Matrix:
+            answer = []
+            for tup in range(len(self.data)):
+                answer.append([])
+                for num in range(len(self.data)):
+                    answer[tup].append(self.data[tup][num] + other[tup][num])
+            sub_list = [tuple(ls) for ls in answer]
+            # convert answer to type Matrix. ls are the inside lists
+            return Matrix((tuple(sub_list)))
+        else:
+            raise SyntaxError("only matrix's can be added together")
+
+    def __sub__(self, other):
```

Comment:
you have other functions that do a very similar functionality, why not use them?

Reply:
i tried:
 answer = self.data.__add__(other.scalar_mul(-1))
            return answer

but got:
negative = self.data.__add__(other.scalar_mul(-1))
TypeError: can only concatenate tuple (not "Matrix") to tuple

Reply:
If you got a bug doesn't mean you shouldn't do it. It mean you need to fix your bug

Reply:
???